### DISCUSSION TITLE
Add container-y class with vertical spacing

### DISCUSSION THREAD

**Discussion by minimit** [tailwindlabs/tailwindcss#2668] [Ideas]
It can be useful sometimes to have a `container-y` class equal to the `container` but with vertical `padding` and `max-height`.

**Comment by simonswiss** [comment]
Hey 👋

I have personally never reached out for this, but I can imagine there are plenty of situations where it could be useful.

There is probably not enough demand for this to be adding it in Tailwind core, but this would be an excellent use case for a plugin that injects a `container-y` class in the `@tailwind components` layer  with [`addComponents`](https://tailwindcss.com/docs/plugins#adding-components) 👍

**Comment by kevinruscoe** [comment]
As theme.container.padding takes 1 value and applies it horizontally, I think we could easily just allow an actual css padding value to achieve part of this.

```
theme.container.padding: "1rem" => padding-left: 1rem; padding-right: 1rem;
theme.container.padding: "1rem 2rem" => padding: 1rem 2rem;
theme.container.padding: "1rem 2rem 3rem" => padding: 1rem 2rem 3rem;
theme.container.padding: "1rem 2rem 3rem 4rem" => padding: 1rem 2rem 3rem 4rem;
```

**Comment by minimit** [comment]
I'm using padding utilities maybe it's better
